politeiad/backendv2/tstorebe/inventory.go

```diff
@@ -44,12 +44,22 @@ func (t *tstoreBackend) invPathVetted() string {
 }
 
 // invRemoveUnvetted removes the unvetted inventory from its respective path.
+//
+// This function must be called WITHOUT the write lock held.
 func (t *tstoreBackend) invRemoveUnvetted() error {
+	t.RLock()
```

Comment:
You need to apply the full lock since you're deleting data, not just the read lock.